### PR TITLE
Feature/sg 1172 criterion params removal

### PR DESCRIPTION
**Pull request by shaydeci**
- Added decorator for deprecating parameters on top of TrainingParams override method.
- Deprecated criterion_params and removed their usages.

### CODE REVIEW

File: src/super_gradients/recipes/imagenet_resnet50_kd.yaml [hunk 1]
```diff
@@ -25,11 +25,11 @@ val_dataloader: imagenet_val
...
-  loss: KDLogitsLoss
-  criterion_params:
-    distillation_loss_coeff: 0.8
-    task_loss_fn:
-      _target_: super_gradients.training.losses.label_smoothing_cross_entropy_loss.LabelSmoothingCrossEntropyLoss
+  loss:
+    KDLogitsLoss:
+      distillation_loss_coeff: 0.8
+      task_loss_fn:
+        _target_: super_gradients.training.losses.label_smoothing_cross_entropy_loss.LabelSmoothingCrossEntropyLoss
```
Comment (BloodAxe) [inline src/super_gradients/recipes/imagenet_resnet50_kd.yaml:32]: Maybe we can also instantiate loss for `task_loss_fn` using LossFactory:

```
loss:
    KDLogitsLoss:
      distillation_loss_coeff: 0.8
      task_loss_fn: LabelSmoothingCrossEntropyLoss

```

It is not directly related to this PR, but caught my attention that it violates our instantiation paradigm

**Review by BloodAxe — DISMISSED** [review]
It's LGTM!
There is one minor comment, leaving up to you whether you want to address it in this PR

File: src/super_gradients/recipes/imagenet_resnet50_kd.yaml (hunk 1 above)
Comment (shaydeci) [reply src/super_gradients/recipes/imagenet_resnet50_kd.yaml]: fixed,

**Review by BloodAxe — APPROVED** [review]
LGTM

**Pull request merged**